File: 100-Days-Series/Day 97/PalindromeArray.java

public class PalindromeArray {
    public static boolean getReverseArray(int arr[]){
        int start=0;
        int end=arr.length-1;
        while(start<end){
            if(arr[start]!=arr[end]){
                return false;
            }
            start++;
            end--;
        }
        return true;
    }
    // Main Method
    public static void main(String [] args){
        int arr[]={1,2,3,2,1};
        System.out.print("Array : ");
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }
        boolean result=getReverseArray(arr);
        if(result){
            System.out.println("\nThe array is a palindrome.\n");
        }else{
            System.out.println("\nThe array is not a palindrome.\n");
        }
    }
}

File: 100-Days-Series/Day 97/PalindromeArray.cpp
#include<iostream>
#include<vector>
using namespace std;
// Function to check if an array is a palindrome.
bool getCheckPalindromeArray(vector<int>&arr){
    int start=0;
    int end=arr.size()-1;
    while(start<end){
        if(arr.at(start)!=arr.at(end)){
            return false;
        }
        start++;
        end--;
    }
    return true;
}
// Let's Start Here
int main(){
    vector<int>arr={1, 2, 3, 2, 1};
    cout<<"\nArray : ";
    for(int i=0;i<arr.size();i++){
        cout<<arr.at(i)<<" ";
    }
    bool result=getCheckPalindromeArray(arr);
    if(result){
        cout<<"\nThe array is a Palindrome."<<endl;
    }else{
        cout<<"\nThe array is not a Palindrome."<<endl;
    }
    return 0;
}